unseen/time_utils.py

```diff
@@ -217,35 +217,53 @@ def get_clim(ds, dims, time_period=None, groupby_init_month=False):
     return clim
 
 
-def select_time_period(ds, period):
+def select_time_period(da, period, time_name="time"):
```

Comment:
My general approach throughout the package has been to assume the dimension names `time`, `lat`, `lon`, `init_date` and `lead_time` to avoid having to define keyword arguments like `time_name`, `lat_name` etc for every function. (The input dimension names are renamed if necessary by passing a configuration file to `fileio.open_file`.)

We could switch to an approach of having keyword arguments for dimension names in functions - we just need to make sure to be consistent across all modules.